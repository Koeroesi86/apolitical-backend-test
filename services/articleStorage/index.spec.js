const moxios = require("moxios");
const articleStorage = require("./index");

describe('articleStorage', () => {
  beforeEach(() => {
    moxios.install();
  });

  afterEach(() => {
    moxios.uninstall();
  });

  it('should fetch articles', async () => {
    moxios.stubRequest('http://asdfast.beobit.net/api/', {
      status: 200,
      response: {
        text: 'Lorem ipsum'
      }
    });

    const res = await articleStorage();

    expect(moxios.requests.mostRecent()).toMatchSnapshot('request');
    expect(res).toMatchSnapshot('result');
  })
})