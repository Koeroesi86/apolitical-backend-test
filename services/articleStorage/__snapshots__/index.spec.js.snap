// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`articleStorage should fetch articles: request 1`] = `
Request {
  "config": Object {
    "adapter": [Function],
    "data": undefined,
    "headers": Object {
      "Accept": "application/json, text/plain, */*",
    },
    "maxBodyLength": -1,
    "maxContentLength": -1,
    "method": "get",
    "timeout": 0,
    "transformRequest": Array [
      [Function],
    ],
    "transformResponse": Array [
      [Function],
    ],
    "url": "http://asdfast.beobit.net/api/",
    "validateStatus": [Function],
    "xsrfCookieName": "XSRF-TOKEN",
    "xsrfHeaderName": "X-XSRF-TOKEN",
  },
  "headers": Object {
    "Accept": "application/json, text/plain, */*",
  },
  "reject": [Function],
  "resolve": [Function],
  "responseType": undefined,
  "timeout": 0,
  "url": "http://asdfast.beobit.net/api/",
  "withCredentials": false,
}
`;

exports[`articleStorage should fetch articles: result 1`] = `"Lorem ipsum"`;
